src/transactions/feed/TransactionFeedItemImage.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import ContactCircle from 'src/components/ContactCircle'
+import GreenLoadingSpinner from 'src/icons/GreenLoadingSpinner'
+import SwapIcon from 'src/icons/SwapIcon'
+import { Recipient } from 'src/recipients/recipient'
+import { TransactionStatus } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+
+const AVATAR_SIZE = 40
+
+type Props =
+  | {
+      status: TransactionStatus
+      __typename: 'TokenExchangeV3'
+    }
+  | {
+      status: TransactionStatus
+      __typename: 'TokenTransferV3'
+      recipient: Recipient
+    }
```

Comment:
I think it may be the first time we really use the `__typename` field.
Do we still wanna use `__typename` here for the prop name? It it gives the feeling it's a private thing that shouldn't be used.

Also don't we want NFTs here too?

Reply:
omg yeah i felt uneasy about this too! the underscores are really awkward, i renamed the prop to `transactionType`

Reply:
i didn't put NFTs here for a few reasons...i'm lazy and because we don't yet have any paths where an NFT transaction could be pending or failed (since you can only receive) then the NFTs don't really benefit from this component. also there's a bit of logic about showing the NFT image (rendering the image vs fallback etc) which would pollute this component a bit